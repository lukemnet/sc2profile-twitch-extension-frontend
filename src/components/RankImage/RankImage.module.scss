@mixin rankImageDimensions {
  width: 47px;
  height: 53px;
}

@mixin rankImage($rank, $bgPositionX) {
  &.#{$rank}_1:after,
  &.#{$rank}_2:after,
  &.#{$rank}_3:after,
  &.#{$rank}_4:after {
    background-position-x: $bgPositionX;
  }

  &.#{$rank}_1:after {
    background-position-y: -145px;
  }


  &.#{$rank}_2:after {
    background-position-y: -95px;
  }

  &.#{$rank}_3:after {
    background-position-y: -45px;
  }

  &.#{$rank}_4:after {
    background-position-y: 5px;
  }
}

.RankImage {
  @include rankImageDimensions;
  background-image:url('images/frame.png');

  &:after {
    content: '';
    display: block;
    @include rankImageDimensions;
    background:url('images/ranks.png') no-repeat 3px 5px;
  }

  @include rankImage("bronze", -41px);
  @include rankImage("silver", -89px);
  @include rankImage("gold", -137px);
  @include rankImage("platinum", -184px);
  @include rankImage("diamond", -231px);
  @include rankImage("master", -278px);
  @include rankImage("grandmaster", -329px);
}