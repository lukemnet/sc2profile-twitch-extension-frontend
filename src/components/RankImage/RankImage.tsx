import React from 'react';
import classnames from 'classnames/bind';
import calculateRankTier from 'helpers/calculateRankTier/calculateRankTier';
import calculateTierRange from 'helpers/calculateTierRange/calculateTierRange';
import capitalize from 'helpers/capitalize/capitalize';
import { Rank } from 'types';
import styles from './RankImage.module.scss';

interface RankImageProps {
  rank: Rank;
  divisionRank?: number;
}

const cx = classnames.bind(styles);

const RankImage = ({ rank, divisionRank }: RankImageProps) => {
  const tier = calculateRankTier(rank, divisionRank) || 1;
  const tierRange = calculateTierRange(rank, tier)  || 1;
  const title = divisionRank
    ? capitalize(`${rank}, top ${tierRange}`)
    : capitalize(rank);

  return (
    <div
      className={cx('RankImage', `${rank}_${tier}`)}
      title={title}
    />
  );
};

export default RankImage;
