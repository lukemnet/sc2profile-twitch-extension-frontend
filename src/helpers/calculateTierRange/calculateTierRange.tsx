import { Rank, Tier } from 'types';

const calculateTierRange = (rank: Rank, tier?: Tier) => {
  const percentages = [100, 8, 25, 50, 100];
  const multiplier = rank === 'grandmaster' ? 2 : 1;
  return !tier
    ? percentages[0]
    : percentages[tier] * multiplier;
};

export default calculateTierRange;
