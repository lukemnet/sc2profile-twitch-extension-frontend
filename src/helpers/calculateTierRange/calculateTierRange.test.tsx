import { Rank } from 'src/components/RankImage/RankImage';
import calculateTierRange from './calculateTierRange';

const ranks = [
  'bronze',
  'silver',
  'gold',
  'platinum',
  'diamond',
  'master',
  'grandmaster',
] as Rank[];

const tiers = [1, 2, 3, 4, undefined];

it('should calculate correct tier ranges for all ranks and division ranks', () => {
  ranks.map(rank =>
    tiers.map(tier =>
      expect(calculateTierRange(rank, tier)).toMatchSnapshot()
    )
  );
});
