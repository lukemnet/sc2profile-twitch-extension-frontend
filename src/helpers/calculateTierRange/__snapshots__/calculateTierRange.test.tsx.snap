// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should calculate correct tier ranges for all ranks and division ranks 1`] = `8`;

exports[`should calculate correct tier ranges for all ranks and division ranks 2`] = `25`;

exports[`should calculate correct tier ranges for all ranks and division ranks 3`] = `50`;

exports[`should calculate correct tier ranges for all ranks and division ranks 4`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 5`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 6`] = `8`;

exports[`should calculate correct tier ranges for all ranks and division ranks 7`] = `25`;

exports[`should calculate correct tier ranges for all ranks and division ranks 8`] = `50`;

exports[`should calculate correct tier ranges for all ranks and division ranks 9`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 10`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 11`] = `8`;

exports[`should calculate correct tier ranges for all ranks and division ranks 12`] = `25`;

exports[`should calculate correct tier ranges for all ranks and division ranks 13`] = `50`;

exports[`should calculate correct tier ranges for all ranks and division ranks 14`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 15`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 16`] = `8`;

exports[`should calculate correct tier ranges for all ranks and division ranks 17`] = `25`;

exports[`should calculate correct tier ranges for all ranks and division ranks 18`] = `50`;

exports[`should calculate correct tier ranges for all ranks and division ranks 19`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 20`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 21`] = `8`;

exports[`should calculate correct tier ranges for all ranks and division ranks 22`] = `25`;

exports[`should calculate correct tier ranges for all ranks and division ranks 23`] = `50`;

exports[`should calculate correct tier ranges for all ranks and division ranks 24`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 25`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 26`] = `8`;

exports[`should calculate correct tier ranges for all ranks and division ranks 27`] = `25`;

exports[`should calculate correct tier ranges for all ranks and division ranks 28`] = `50`;

exports[`should calculate correct tier ranges for all ranks and division ranks 29`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 30`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 31`] = `16`;

exports[`should calculate correct tier ranges for all ranks and division ranks 32`] = `50`;

exports[`should calculate correct tier ranges for all ranks and division ranks 33`] = `100`;

exports[`should calculate correct tier ranges for all ranks and division ranks 34`] = `200`;

exports[`should calculate correct tier ranges for all ranks and division ranks 35`] = `100`;
